util/chplenv/chpl_aux_filesys.py

```diff
@@ -20,6 +21,24 @@ def get():
             stderr.write("Warning: Can't find your Java installation\n")
         if not found_hdfs:
             stderr.write("Warning: Can't find your Hadoop installation\n")
+
+    elif aux_fs == 'hdfs3':
+        def fetchInfo(env, val, err):
+            stripstr = (env.find('INCLUDE') > 0 and '-I' or '-L')
+            files = map(lambda z: z.lstrip(stripstr), os.environ.get(env, '').split())
+            res = 0
+            for f in files:
+               res += sum(map(lambda x: 1, filter(lambda z: z.find(val) > 0 and 1 or 0, glob("%s/*" % (f,)))))
+
+            if res < 1:
+                stderr.write(err)
+                return False
+
+            return True
+
+        fetchInfo('CHPL_AUXIO_INCLUDE', 'hdfs', "Warning: Can't find your HDFS3 header file installation\n")
+        fetchInfo('CHPL_AUXIO_LIBS', 'hdfs3', "Warning: Can't find your HDFS3 static library installation\n")
```

Comment:
It worries me a little bit that this code is radically different from the Java libhdfs version. Why is it so different? What was different about libhfs3? Are you trying to support some other way of configuring it or something? The map/lambda/sum business seems to obfuscate something but I don't know what yet. I think you could say
that this block of code uses Advanced Python, which I'm not conversant in, and which I don't think is
really called for in this configuration setting.